graphix/pauli.py

```diff
@@ -69,32 +120,32 @@ def prefix(self, s: str) -> str:
             result = "1j*" + s
         else:
             result = s
-        if self.__sign:
+        if self.__sign == Sign.Minus:
             result = "-" + result
         return result
 
     def __mul__(self, other):
         if isinstance(other, ComplexUnit):
             im = self.__im != other.__im
-            sign = (self.__sign != other.__sign) != (self.__im and other.__im)
-            return COMPLEX_UNITS[sign][im]
+            sign = self.__sign * other.__sign * Sign.minus_if(self.__im and other.__im)
+            return COMPLEX_UNITS[sign == Sign.Minus][im]
         return NotImplemented
 
     def __rmul__(self, other):
         if other == 1:
             return self
         elif other == -1:
-            return COMPLEX_UNITS[not self.__sign][self.__im]
+            return COMPLEX_UNITS[self.__sign == Sign.Plus][self.__im]
         elif other == 1j:
-            return COMPLEX_UNITS[self.__sign != self.__im][not self.__im]
+            return COMPLEX_UNITS[self.__sign == Sign.plus_if(self.__im)][not self.__im]
         elif other == -1j:
-            return COMPLEX_UNITS[self.__sign == self.__im][not self.__im]
+            return COMPLEX_UNITS[self.__sign == Sign.minus_if(self.__im)][not self.__im]
 
     def __neg__(self):
-        return COMPLEX_UNITS[not self.__sign][self.__im]
+        return COMPLEX_UNITS[self.__sign == Sign.Plus][self.__im]
 
 
-COMPLEX_UNITS = [[ComplexUnit(sign, im) for im in (False, True)] for sign in (False, True)]
+COMPLEX_UNITS = [[ComplexUnit(sign, im) for im in (False, True)] for sign in (Sign.Plus, Sign.Minus)]
```

Comment:
Let's delete this line to prevent unexpected cache mutation.

Reply:
These global values are needed. I changed them for tuples instead of lists in 57078e7 to prevent unexpected mutations.

Reply:
Why necessary? 
We may need to refactor the class using the singleton pattern.